modtools/change-build-menu.lua

```diff
@@ -0,0 +1,558 @@
+            custom = custom,
+        },
+    })
+end
+
+-- Return early if module mode. Note that the ticker will not start in module mode!
+-- To start the ticker you need to enable the script first ("enable change-build-menu").
+if moduleMode then return end
+
+function usage()
+    dfhack.print [==[
+Change the build sidebar menus.
+
+This script provides a flexible and comprehensive system for adding and removing
+items from the build sidebar menus. You can add or remove workshops/furnaces by
+text ID, or you can add/remove ANY building via a numeric building ID triplet.
+
+Changes made with this script do not survive a save/load. You will need to redo
+your changes each time the world loads.
+
+Just to be clear: You CANNOT use this script AT ALL if there is no world
+loaded!
+
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+
+Command Usage:
+
+    change-build-menu start|enable
+    enable change-build-menu
+        Start the ticker. This needs to be done before any changes will take
+        effect. Note that you can make changes before or after starting the
+        ticker, both options should work equally well.
+    
+    change-build-menu stop|disable
+    disable change-build-menu
+        Stop the ticker. Does not clear stored changes. The ticker will
+        automatically stop when the current world is unloaded.
+    
+    change-build-menu add <ID> <CATEGORY> [<KEY>]
+        Add the workshop or furnace with the ID <ID> to <CATEGORY>.
+        <KEY> is an optional DF hotkey ID.
+        
+        <CATEGORY> may be one of:
+            MAIN_PAGE
+            SIEGE_ENGINES
+            TRAPS
+            WORKSHOPS
+            FURNACES
+            CONSTRUCTIONS
+            MACHINES
+            CONSTRUCTIONS_TRACK
+    
+        Valid <ID> values for hardcoded buildings are as follows:
+            CARPENTERS
+            FARMERS
+            MASONS
+            CRAFTSDWARFS
+            JEWELERS
+            METALSMITHSFORGE
+            MAGMAFORGE
+            BOWYERS
+            MECHANICS
+            SIEGE
+            BUTCHERS
+            LEATHERWORKS
+            TANNERS
+            CLOTHIERS
+            FISHERY
+            STILL
+            LOOM
+            QUERN
+            KENNELS
+            ASHERY
+            KITCHEN
+            DYERS
+            TOOL
+            MILLSTONE
+            WOOD_FURNACE
+            SMELTER
+            GLASS_FURNACE
+            MAGMA_SMELTER
+            MAGMA_GLASS_FURNACE
+            MAGMA_KILN
+            KILN
+    
+    change-build-menu remove <ID> <CATEGORY>
+        Remove the workshop or furnace with the ID <ID> from <CATEGORY>.
+        
+        <CATEGORY> and <ID> may have the same values as for the "add" option.
+    
+    change-build-menu revert <ID> <CATEGORY>
+        Revert an earlier remove or add operation. It is NOT safe to "remove"
+        and "add"ed building or vice versa, use this option to reverse any

```

Comment:
Oops...

Actually I was reviewing the offline master copy this morning (backporting changes I had made to the DFHack copy) and noticed that myself. Thank you for reminding me though, or I would have forgotten to change it in the online copy.